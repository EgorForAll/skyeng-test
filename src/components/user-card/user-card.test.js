import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import UserCard from "./user-card";

const userMock = {
  login: "Brad",
  id: 1,
  node_id: "",
  avatar_url: "https://brad-avatar.ru",
  gravatar_id: "",
  url: "",
  html_url: "https://brad-page.git",
  followers_url: "",
  following_url: "",
  gists_url: "",
  starred_url: "",
  subscriptions_url: "",
  organizations_url: "",
  repos_url: "",
  events_url: "",
  received_events_url: "",
  type: "User",
  site_admin: false,
  score: 2,
};

describe("UserCard", () => {
  it("UserCard component schould render correctly", () => {
    render(<UserCard user={userMock} />);
    expect(screen.getByRole("button", { name: "Узнать тип пользователя" }));
    expect(screen.getByText("Профиль:"));
    expect(screen.getByText(userMock.login));
    expect(screen.getByRole("link")).toHaveAttribute("href", userMock.html_url);
    expect(screen.getByRole("img")).toHaveAttribute("src", userMock.avatar_url);
  });
  it("UserCard snapshot", () => {
    const userCard = render(<UserCard user={userMock} />);
    expect(userCard).toMatchSnapshot();
  });
});
