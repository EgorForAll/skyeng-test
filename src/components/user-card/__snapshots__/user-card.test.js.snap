// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard UserCard snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="user-card"
      >
        <h2
          class="user-name"
        >
          Brad
        </h2>
        <img
          alt="аватар пользователя"
          height="200"
          src="https://brad-avatar.ru"
          width="200"
        />
        <span
          class="user-id"
        >
          id: 
          1
        </span>
        <div
          class="profile-wrapper"
        >
          <span>
            Профиль:
             
            <a
              class="card-link"
              href="https://brad-page.git"
            >
              https://brad-page.git
            </a>
          </span>
        </div>
        <button
          class="card-btn"
        >
          Узнать тип пользователя
        </button>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="user-card"
    >
      <h2
        class="user-name"
      >
        Brad
      </h2>
      <img
        alt="аватар пользователя"
        height="200"
        src="https://brad-avatar.ru"
        width="200"
      />
      <span
        class="user-id"
      >
        id: 
        1
      </span>
      <div
        class="profile-wrapper"
      >
        <span>
          Профиль:
           
          <a
            class="card-link"
            href="https://brad-page.git"
          >
            https://brad-page.git
          </a>
        </span>
      </div>
      <button
        class="card-btn"
      >
        Узнать тип пользователя
      </button>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
